.buildkite/auditbeat/auditbeat-pipeline.yml

```diff
@@ -19,6 +19,10 @@ env:
   IMAGE_WIN_2019: "family/platform-ingest-beats-windows-2019"
   IMAGE_WIN_2022: "family/platform-ingest-beats-windows-2022"
 
+  ARCH_UNIX_X86_64: "-arch amd64"
+  ARCH_MINGW_X86_64: "-arch 386"
+  ARCH_UNIX_ARM_64: "-arch arm"
```

Comment:
I understand that we want to be as explicit as possible and not have "magic" scripts.
However, I fear that that the magnitude of repetition of the above env vars (not just the declaration, but their use too) make it extremely easy to forget a definition, and "bloat" the pipeline yaml files unnecessarily.

My suggestion is:

Instead, add a very simple conditional in the existing [pre-command repository script](https://github.com/elastic/beats/blob/main/.buildkite/hooks/pre-command#L33):

```
CPU_ARCH=$(uname -m)
PLATFORM_TYPE=$(uname)

if [ "${CPU_ARCH}" == "x86_64" ]; then
  case "${PLATFORM_TYPE}" in
    Linux|Darwin)
      export GOX_FLAGS="-arch amd64"
      ;;
    MINGW*)
      export GOX_FLAGS="-arch 386"
      ;;
  esac
elif [[ "${CPU_ARCH}" == "aarch64" || "${CPU_ARCH}" == "arm64" ]]; then
  export GOX_FLAGS="-arch arm"
else
  echo "Unsupported OS"
  exit 1
fi
```

To me, it sounds like an ideal use of Buildkite's [pre-command repository hook](https://buildkite.com/docs/agent/v3/hooks#hook-locations-repository-hooks) feature, and we already have a short beats [pre-command hook](https://github.com/elastic/beats/blob/main/.buildkite/hooks/pre-command). For manual reproductions, our [instructions](https://docs.elastic.dev/ingest-dev-docs/beats/beats-ci#reproducing-pull-request-failures) already document sourcing both hooks (agent and repository level).

With this suggestion, we can remove all references to the above three variables and env definitions per step, which will reduce immensely the size of this PR and potential for future typos and omissions.

WDYT?

Reply:
Agree, moved GOX_FLAGS definition to pre-command hook. And removed this env var definition frominstall-macos-tools.sh.